plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -862,6 +863,110 @@ public void testTimestampFromTrino(ZoneId sessionZone)
                 .execute(getQueryRunner(), session, trinoCreateAndInsert("test_timestamp"));
     }
 
+    @Test
+    public void testTimestampWithTimeZoneFromTrinoUtc()
+    {
+        ZoneId sessionZone = UTC;
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        SqlDataTypeTest.create()
+                // after epoch (MySQL's timestamp type doesn't support values <= epoch)
+                .addRoundTrip("timestamp(3) WITH TIME ZONE", "TIMESTAMP '2019-03-18 10:01:17.987 %s'".formatted(sessionZone), createTimestampWithTimeZoneType(3), "TIMESTAMP '2019-03-18 10:01:17.987 UTC'")
```

Comment:
add a separate test method verifying that the failure happens as expected (e..g it's a good user facing error).